core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.kafka.scaladsl
+
+import scala.concurrent.Future
+import akka.Done
+import akka.stream.scaladsl.Source
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import akka.kafka.internal.ConsumerStage
+import akka.NotUsed
+import akka.dispatch.ExecutionContexts
+import scala.concurrent.ExecutionContext
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.clients.consumer.OffsetAndMetadata
+import akka.kafka.internal.ConsumerStage.CommittableOffsetBatchImpl
+
+object Consumer {
+
+  trait Message[K, V] {
+    def key: K
+    def value: V
+    def offset: Offset
+  }
+
+  final case class CommittableMessage[K, V](
+      key: K,
+      value: V,
+      committableOffset: CommittableOffset
+  ) extends Message[K, V] {
+
+    override def offset: Offset = committableOffset.offset
+  }
+
+  /**
+   * This interface might move into `akka.stream`
+   */
+  trait Committable {
+    def commit(): Future[Done]
+  }
+
+  trait CommittableOffset extends Committable {
+    def offset: Offset
+  }
+
+  final case class Offset(topic: String, partition: Int, offset: Long)
+
+  object CommittableOffsetBatch {
+    val empty: CommittableOffsetBatch = CommittableOffsetBatchImpl(Map.empty)(None)
+  }
+
+  trait CommittableOffsetBatch extends Committable {
+    def updated(offset: CommittableOffset): CommittableOffsetBatch
+
+    def offsets: Map[TopicPartition, OffsetAndMetadata]
+  }
+
+  trait Control {
+    def stop(): Future[Done]
+    def stopped: Future[Done]
+  }
+
+  /**
+   * No support for committing offsets to Kafka. Can be used when offset is stored externally or with auto-commit.
+   */
+  def plainSource[K, V](consumerProvider: () => KafkaConsumer[K, V]): Source[Message[K, V], Control] =
```

Comment:
Ah. Ok, this factory method is a bit confusing. Can't we have a source version that does not expose the factory method but creates a consumer itself?